app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_user_posts_counter
+    user.update(posts_counter: user.posts.count)
+  end
+
+  def five_most_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
```

Comment:
- [ Optional] It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private 👍 

````ruby
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
````

When you make functions private, you prevent the function from being accessible by others and that way, you prevent malicious attacks. 👍